seqr/views/apis/anvil_workspace_api.py

```diff
@@ -1,57 +1,133 @@
+    :param namespace: The namespace (or the billing account) of the workspace
+    :param name: The name of the workspace. It also be used as the project name
+    :return the projectsByGuid with the new project json
 
     """
-    response_json = {}  # to be done
-    # Todo:
-    # 1) Validate that the current user has logged in through google and has one of the valid can_edit levels of
-    #  access on the specified workspace;
-    # 2) Validate all the user input from the post body;
-    # 3) Add the seqr service account to the corresponding AnVIL workspace, so that our team will have access to the
-    # project for data loading;
-    # 4) Create a new Project in seqr. This project should NOT be added to the analyst group. The project name should
-    #  just be the workspace name. Make sure to set workspace_namespace and workspace_name correctly;
-    # 5) Add families/individuals based on the uploaded pedigree file;
-    # 6) Send an email to all seqr data managers saying a new AnVIL project is ready for loading. Include the seqr
-    #  project guid, the workspace name, and attach a txt file with a list of the individual IDs that were created.
-    return create_json_response(response_json)
+    project_name = _get_project_name(namespace, name)
+    project = Project.objects.filter(name = project_name)
+    if project:
+        error = 'Project {} exists.'.format(project_name)
+        logger.warning(error)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Validate that the current user has logged in through google and has sufficient permissions
+    if not workspace_has_perm(request.user, CAN_EDIT, namespace, name, can_share=True):
+        error = "Missing required permissions for loading data from workspace {}/{}".format(namespace, name)
+        logger.warning(error)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Validate all the user inputs from the post body
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['genomeVersion', 'uploadedFileId'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not request_json.get('agreeSeqrAccess'):
+        error = 'Must agree to grant seqr access to the data in the associated workspace.'
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Add the seqr service account to the corresponding AnVIL workspace
+    add_service_account(request.user, namespace, name)
+
+    # Parse families/individuals in the uploaded pedigree file
+    try:
+        json_records = load_uploaded_file(request_json['uploadedFileId'])
+    except Exception as ee:
+        error = "Uploaded pedigree file is missing or other exception: {}".format(str(ee))
```

Comment:
its not missing, you checked that above. This should just be `'Error with uploaded pedigree file: {}'.format(e)`

Reply:
The function `load_uploaded_file()` deletes the temp file after loading it. I encountered errors during debugging because the file has been deleted. The exception message is confusing. Therefore, I add a more clear indication. You are right, it is not always missing. Maybe I should say `'Error with uploaded pedigree file. Try re-uploading it. Error information: {}'.format(e)`.